synapse/models/syn.py

```diff
@@ -197,3 +198,35 @@ def _revModl201710191144(self):
                     i = i + len(chunk)
                     logger.debug('Loading {:,d} [{}%] rows into transaction'.format(i, int((i / tot) * 100)))
         logger.debug('Finished adding node:created rows to the Cortex')
+
+    @modelrev('syn', 201711012123)
+    def _revModl201711012123(self):
+        with self.core.getCoreXact():
+            now = s_common.now()
+            forms = set()
+            adds = []
+            # Lifting all tufo:form rows first and then grabbing the unique
+            # forms referenced allows us to migrate forms which are not loaded
+            # at the time this migration is performed (important in the event
+            # a custom model is loaded) after the Cortex is started.
+            logger.debug('Lifting tufo:form rows')
+            for i, p, v, t in self.core.store.getRowsByProp('tufo:form'):
```

Comment:
this will cause a *huge* lift.  can we do this via the model defs?

Reply:
I thought about this and we actually cannot guarantee that we'll have models loaded during the execution of the migration.  If an application spun up a cortex and THEN added in a coremodule after the fact, they would not be able to benefit from this ndef migration and would have to write their own migration code after-the-fact.  This migrates iden with an associated tufo:form value.